lib/json/common.rb

```diff
@@ -642,12 +639,14 @@ def self.iconv(to, from, string)
     string.encode(to, from)
   end
 
-  private
```

Comment:
`merge_dump_options` should be private instance method and also private module function
```ruby
module_function

def a_private_instance_method_and_public_module_function
end

private # module_function rule is cleared

def a_private_instance_method # not a module function
end
```